internal/dwarf/frame/table.go

```diff
@@ -531,135 +537,150 @@ func restoreextended(frameContext *FrameContext) {
 	)
 
 	frame.CFA.Expression = expr
 	frame.CFA.Rule = RuleExpression
 }
 
-func expression(frameContext *FrameContext) {
-	frame := frameContext.getCurrentInstruction()
+func expression(ctx *Context) {
+	frame := ctx.getCurrentInstruction()
 
 	var (
-		reg, _ = util.DecodeULEB128(frameContext.buf)
-		l, _   = util.DecodeULEB128(frameContext.buf)
-		expr   = frameContext.buf.Next(int(l))
+		reg, _ = util.DecodeULEB128(ctx.buf)
+		l, _   = util.DecodeULEB128(ctx.buf)
+		expr   = ctx.buf.Next(int(l))
 	)
 
 	// fmt.Println("register", reg, "set to expression")
 	frame.Regs[reg] = DWRule{Rule: RuleExpression, Expression: expr}
 }
 
-func offsetextendedsf(frameContext *FrameContext) {
-	frame := frameContext.getCurrentInstruction()
+func offsetextendedsf(ctx *Context) {
+	frame := ctx.getCurrentInstruction()
 
 	var (
-		reg, _    = util.DecodeULEB128(frameContext.buf)
-		offset, _ = util.DecodeSLEB128(frameContext.buf)
+		reg, _    = util.DecodeULEB128(ctx.buf)
+		offset, _ = util.DecodeSLEB128(ctx.buf)
 	)
 
 	// fmt.Println("register", reg, "set to dwrule")
 	frame.Regs[reg] = DWRule{Offset: offset * frame.dataAlignment, Rule: RuleOffset}
 }
 
-func valoffset(frameContext *FrameContext) {
-	frame := frameContext.getCurrentInstruction()
+func valoffset(ctx *Context) {
+	frame := ctx.getCurrentInstruction()
 
 	var (
-		reg, _    = util.DecodeULEB128(frameContext.buf)
-		offset, _ = util.DecodeULEB128(frameContext.buf)
+		reg, _    = util.DecodeULEB128(ctx.buf)
+		offset, _ = util.DecodeULEB128(ctx.buf)
 	)
 
 	// fmt.Println("register", reg, "set to dwrule")
 	frame.Regs[reg] = DWRule{Offset: int64(offset), Rule: RuleValOffset}
 }
 
-func valoffsetsf(frameContext *FrameContext) {
-	frame := frameContext.getCurrentInstruction()
+func valoffsetsf(ctx *Context) {
+	frame := ctx.getCurrentInstruction()
 
 	var (
-		reg, _    = util.DecodeULEB128(frameContext.buf)
-		offset, _ = util.DecodeSLEB128(frameContext.buf)
+		reg, _    = util.DecodeULEB128(ctx.buf)
+		offset, _ = util.DecodeSLEB128(ctx.buf)
 	)
 
 	// fmt.Println("register", reg, "set to offset")
 	frame.Regs[reg] = DWRule{Offset: offset * frame.dataAlignment, Rule: RuleValOffset}
 }
 
-func valexpression(frameContext *FrameContext) {
-	frame := frameContext.getCurrentInstruction()
+func valexpression(ctx *Context) {
+	frame := ctx.getCurrentInstruction()
 
 	var (
-		reg, _ = util.DecodeULEB128(frameContext.buf)
-		l, _   = util.DecodeULEB128(frameContext.buf)
-		expr   = frameContext.buf.Next(int(l))
+		reg, _ = util.DecodeULEB128(ctx.buf)
+		l, _   = util.DecodeULEB128(ctx.buf)
+		expr   = ctx.buf.Next(int(l))
 	)
 
 	frame.Regs[reg] = DWRule{Rule: RuleValExpression, Expression: expr}
 }
 
-func louser(frameContext *FrameContext) {
-	frameContext.buf.Next(1)
+func louser(ctx *Context) {
+	ctx.buf.Next(1)
 }
 
-func hiuser(frameContext *FrameContext) {
-	frameContext.buf.Next(1)
+func hiuser(ctx *Context) {
+	ctx.buf.Next(1)
 }
 
-func gnuargsize(frameContext *FrameContext) {
```

Comment:
cc @javierhonduco

Reply:
This is what confused me, thanks! I see the implementation doesn't change for this opcode, as we were already reading the unsigned LEB128, which makes sense 😄